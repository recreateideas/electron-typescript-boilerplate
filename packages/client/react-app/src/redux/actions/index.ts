import * as common from './common/actions';
import * as router from './router/actions';

export { common, router };
