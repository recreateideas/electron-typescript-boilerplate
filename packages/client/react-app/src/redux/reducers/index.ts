import common from './common';

const reducers = {
    common,
};

export default reducers;
