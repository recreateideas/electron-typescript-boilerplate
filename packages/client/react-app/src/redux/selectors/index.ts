import * as common from './common';
import * as router from './router';

export { common, router };
