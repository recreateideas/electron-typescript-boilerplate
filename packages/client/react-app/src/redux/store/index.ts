import initialState from './initialState';
import store from './store';

export * from './declarations.d';
export { initialState };
export default store;
