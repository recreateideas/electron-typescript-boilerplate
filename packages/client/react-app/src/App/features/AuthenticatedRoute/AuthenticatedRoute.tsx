import { memo, ReactElement, useEffect, useLayoutEffect } from 'react';
import { LocationDescriptor } from 'history';
import { Route, Redirect } from 'react-router-dom';
import { isEqual } from 'lodash';
import { actions, selectors, useSelector, useDispatch } from '../../../redux';
// import { Header } from '..';

// redirectUrl: PropTypes.string,
// skipGetUser: PropTypes.bool,
// showBackButton: PropTypes.bool,
// title: PropTypes.string,
// contextRoot: PropTypes.string,
// };

interface IProps {
    redirectUrl?: LocationDescriptor<unknown>;
    title?: string;
    showBackButton?: boolean;
    exact?: boolean;
    path?: string | string[];
    location?: {
        pathname?: string;
    };
    render: (props: any) => ReactElement;
}

export const AuthenticatedRoute = memo((props: IProps) => {
    const { redirectUrl, title = 'Random', showBackButton, ...routeProps } = props;
    const {
        router: { setRouteParams },
    } = actions;
    const dispatch = useDispatch();
    const { router: routerSelectors } = selectors;
    const location = useSelector(routerSelectors.location);
    const isAuthenticated = true;

    const { location: locationProps, path } = routeProps;
    useEffect(() => {
        if (path && locationProps?.pathname) {
            dispatch(setRouteParams(path));
        }
    }, [dispatch, locationProps, path, setRouteParams]);

    useLayoutEffect(() => {
        document.title = `Stacks - ${title}`;
    }, [title]);

    const allProps = {
        ...routeProps,
        location,
    };
    return isAuthenticated ? (
        <>
            {/* <Header {...{ showBackButton }} /> */}
            <Route {...allProps} />
        </>
    ) : (
        /* istanbul ignore next */
        <Redirect to={redirectUrl as LocationDescriptor} />
    );
}, isEqual);

AuthenticatedRoute.displayName = 'AuthenticatedRoute';
