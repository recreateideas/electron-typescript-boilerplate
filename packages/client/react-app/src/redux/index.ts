import { useDispatch } from 'react-redux';
import * as selectors from './selectors';
import * as actions from './actions';
import { useSelector } from './custom-hooks';

export * from './store';

export { useDispatch, actions, useSelector, selectors };
