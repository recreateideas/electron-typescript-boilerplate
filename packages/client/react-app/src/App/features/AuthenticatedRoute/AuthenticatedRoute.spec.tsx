import { AuthenticatedRoute } from './AuthenticatedRoute';
jest.mock('../../../redux');
import * as redux from '../../../redux';
import { TestProvider } from '../../../tests';

const { mountWithRouterProps } = TestProvider({});

describe('AuthenticatedRoute', () => {
    it('should render', () => {
        const props = {
            render: () => <div />,
        };
        const state = {
            router: {
                location: 'some-location',
            },
        };
        const root = mountWithRouterProps(<AuthenticatedRoute {...props} />, state);
        expect(root).toMatchSnapshot();
    });

    it('should call dispatch setRouteParams if path and location props are supplied', () => {
        const props = {
            render: () => <div />,
            path: 'some-path',
            location: {
                pathname: 'some-path-name',
            },
        };
        const state = {
            router: {
                location: 'some-location',
            },
        };
        const {
            actions: { router },
        } = redux;
        const actionSpy = jest.fn();
        const dispatchSpy = jest.fn();
        jest.spyOn(router, 'setRouteParams').mockReturnValue(actionSpy);
        jest.spyOn(redux, 'useDispatch').mockReturnValue(dispatchSpy);
        mountWithRouterProps(<AuthenticatedRoute {...props} />, state);
        expect(dispatchSpy).toHaveBeenCalledWith(actionSpy);
    });

    it('should NOT call dispatch setRouteParams if no path and location props are supplied', () => {
        const props = {
            render: () => <div />,
        };
        const state = {
            router: {
                location: 'some-location',
            },
        };
        const dispatchSpy = jest.fn();
        jest.spyOn(redux, 'useDispatch').mockReturnValue(dispatchSpy);
        mountWithRouterProps(<AuthenticatedRoute {...props} />, state);
        expect(dispatchSpy).not.toHaveBeenCalled();
    });
});
