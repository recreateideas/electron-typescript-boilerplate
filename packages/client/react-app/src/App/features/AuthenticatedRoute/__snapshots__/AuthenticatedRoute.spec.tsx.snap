// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedRoute should render 1`] = `<div />`;
